From ae162b72f8ffadcb2a5bd2fd4833da815b6c51a1 Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Wed, 29 May 2024 06:22:15 +0000
Subject: [PATCH] ipc-shm: Drop the THIS_MODULE argument from class_create()

Due to the change introduced by upstream commit dcfbb67e48a2 ("driver core:
class: use lock_class_key already present in struct subsys_private"), we need
to drop the THIS_MODULE argument from class_create().

Upstream-Status: Pending

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 os_kernel/ipc-cdev.c | 2 +-
 os_kernel/ipc-uio.c  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/os_kernel/ipc-cdev.c b/os_kernel/ipc-cdev.c
index 2458ae5..cf168c4 100644
--- a/os_kernel/ipc-cdev.c
+++ b/os_kernel/ipc-cdev.c
@@ -314,7 +314,7 @@ static int ipc_cdev_init(void)
 		return err;
 	}
 	/* Create class and init character device */
-	ipc_cdev_priv.ipc_class = class_create(THIS_MODULE, DEVICE_NAME);
+	ipc_cdev_priv.ipc_class = class_create(DEVICE_NAME);
 	cdev_init(&ipc_cdev_priv.ipc_cdev, &ipc_cdev_fops);
 	ipc_cdev_priv.ipc_cdev.owner = THIS_MODULE;
 	cdev_add(&ipc_cdev_priv.ipc_cdev, ipc_cdev_priv.dev_major_num, 1);
diff --git a/os_kernel/ipc-uio.c b/os_kernel/ipc-uio.c
index 4e888f7..d96199f 100644
--- a/os_kernel/ipc-uio.c
+++ b/os_kernel/ipc-uio.c
@@ -296,7 +296,7 @@ static int ipc_shm_uio_probe(struct platform_device *pdev)
 		shm_err("Failed to add cdev\n");
 		goto fail_add_cdev;
 	}
-	ipc_pdev_priv.cdev_class = class_create(THIS_MODULE, IPC_CDEV_NAME);
+	ipc_pdev_priv.cdev_class = class_create(IPC_CDEV_NAME);
 	if (!ipc_pdev_priv.cdev_class) {
 		err = -EEXIST;
 		shm_err("Failed to create class\n");
-- 
2.34.1

